app/src/main/java/com/x8bit/bitwarden/data/auth/repository/util/AuthDiskSourceExtensions.kt

```diff
@@ -183,8 +184,39 @@ val AuthDiskSource.onboardingStatusChangesFlow: Flow<OnboardingStatus?>
         }
         .distinctUntilChanged()
 
+/**
+ * Returns the current [OnboardingStatus] of the active user.
+ */
 val AuthDiskSource.currentOnboardingStatus: OnboardingStatus?
     get() = this
         .userState
         ?.activeUserId
         ?.let { this.getOnboardingStatus(userId = it) }
+
+/**
+ * Returns a [Flow] that emits every time the active user's first time state is changed.
+ */
+@OptIn(ExperimentalCoroutinesApi::class)
+val AuthDiskSource.firstTimeStateFlow: Flow<UserState.FirstTimeState>
+    get() = activeUserIdChangesFlow
+        .flatMapLatest { activeUserId ->
+            combine(
+                listOf(
+                    activeUserId
+                        ?.let {
+                            getShowImportLoginsFlow(it)
+                        }
+                        ?: flowOf(null),
+                ),
+            ) {
+                UserState.FirstTimeState(
+                    showImportLoginsCoachMarker = it[0],
+                )
+            }
+        }
+        .distinctUntilChanged()
+
+val AuthDiskSource.defaultUserFirstTimeState
+    get() = UserState.FirstTimeState(
```

Comment:
Wait, I thought this would pull from the disk source:

```kotlin
this
        .userState
        ?.activeUserId
        ?.let {
            UserState.FirstTimeState(
                showImportLoginsCard = getShowImportLogins(it),
            )
        }
```

Reply:
oh duh

Reply:
I took it very literal